synapse/rest/client/devices.py

```diff
@@ -259,18 +263,100 @@ class DehydratedDeviceServlet(RestServlet):
       "device_id": "dehydrated_device_id"
     }
 
-    """
+    GET /org.matrix.msc3814.v1/dehydrated_device
+    HTTP/1.1 200 OK
 
-    PATTERNS = client_patterns("/org.matrix.msc2697.v2/dehydrated_device$", releases=())
+    Content-Type: application/json
 
-    def __init__(self, hs: "HomeServer"):
+    {
+      "device_id": "dehydrated_device_id",
+      "device_data": {
+        "algorithm": "org.matrix.msc2697.v1.dehydration.v1.olm",
+        "account": "dehydrated_device"
+      }
+    }
+
+    PUT /org.matrix.msc3814.v1/dehydrated_device
+    Content-Type: application/json
+
+    {
+        "device_id": "dehydrated_device_id",
```

Comment:
Confusing---the MSC doesn't have a device_id field on `PUT /dehydrated_device`.

Reply:
I explained the reasoning here somewhat: https://github.com/matrix-org/synapse/pull/15929#issuecomment-1641890467.

I think using the public key of the device as the device id makes sense here. It's probably something we should be doing across the board from the start for `/login` as well, considering the various traps we have because a device ID can be assigned to differing device keys.